src/v/kafka/server/connection_context.cc

```diff
@@ -197,13 +197,13 @@ ss::future<ss::semaphore_units<>>
 connection_context::reserve_request_units(size_t size) {
     // Allow for extra copies and bookkeeping
     auto mem_estimate = size * 2 + 8000; // NOLINT
-    if (mem_estimate >= (size_t)std::numeric_limits<int32_t>::max()) {
-        // TODO: Create error response using the specific API?
-        throw std::runtime_error(fmt::format(
-          "request too large > 1GB (size: {}; estimate: {})",
-          size,
-          mem_estimate));
-    }
```

Comment:
I don't think this is related to #2879, but rather it is checking that a incoming request from a Kafka client does not use too much memory. Am I missing something else here?

Reply:
Yes. No, you are correct.